File: JavaLibrary/src/main/java/com/flat/simplex/parser/logic/line/LineOp.java
package com.flat.simplex.parser.logic.line;

import com.flat.simplex.lexer.Key;
import com.flat.simplex.lexer.Token;
import com.flat.simplex.parser.logic.Block;

public class LineOp extends Line {

    public static final int SetPrecedence = 13;

    public LineOp(Block parent, Token token) {
        super(parent, token);
    }

    public Key getKey() {
        return token.getKey();
    }

    public int getPrecedence() {
        return token.getKey().op;
    }

    @Override
    public boolean isOp() {
        return true;
    }

    @Override
    public LineOp getOp() {
        return this;
    }

    @Override
    public String toString() {
        return "" + token;
    }
}

File: SimplexApp/simplex/lang/parser/logic/line/LineOp.js
const Line = require("simplex/lang/parser/logic/line/Line.js");

class LineOp extends Line {
    constructor(parent, token) {
        super(parent, token);
    }
    getKey() {
        return this.token.getKey();
    }
    getPrecedence() {
        return this.token.getKey().op;
    }
    isOp() {
        return true;
    }
    getOp() {
        return this;
    }
    toString() {
        return '' + this.token;
    }
}
LineOp.SetPrecedence = 13;

module.exports = LineOp;